core/server/master/src/main/java/alluxio/master/meta/checkconf/ServerConfigurationChecker.java

```diff
@@ -60,8 +62,8 @@ public ServerConfigurationChecker(ServerConfigurationStore masterStore,
     mMasterStore = masterStore;
     mWorkerStore = workerStore;
     mConfigCheckReport = new ConfigCheckReport();
-    mMasterStore.registerChangeListener(this::regenerateReport);
-    mWorkerStore.registerChangeListener(this::regenerateReport);
+    mMasterStore.registerChangeListener(() -> mConfigDirty = true);
+    mWorkerStore.registerChangeListener(() -> mConfigDirty = true);
```

Comment:
Does this listener need to be synchronized?

Reply:
no, the variable is volatile

Reply:
done